app/models/user.rb

```diff
@@ -1,4 +1,29 @@
 class User < ActiveRecord::Base
   has_many :bids
   validates :email, email: true, allow_blank: true
+  before_save :clear_sam_status_if_duns_changed
```

Comment:
@baccigalupi I agree, these are the changes we want. I'm have a WIP PR open here now https://github.com/18F/micropurchase/pull/196 containing my first sketches at implementing these. I don't think we're going to let that block the merging of this PR though, since we need to get some things ready for a Monday auction launch.